Framework/Algorithms/src/CreateWorkspace.cpp

```diff
@@ -199,13 +199,6 @@ void CreateWorkspace::exec() {
     const std::vector<double>::difference_type yEnd = yStart + ySize;
     auto local_i = localIndices[0];
 
-    if (!parentWS) {
-      // There is no instrument so setting detector IDs makes no sense, but if
-      // we don't we break quite a few unit tests relying on this legacy
-      // behavior.
-      outputWS->getSpectrum(local_i).setDetectorID(i + 1);
```

Comment:
Good to see this gone. Could you modify the unit tests for `CreateWorkspace` to guard against its accidental/deliberate reintroduction